crates/nu-cli/src/repl.rs

```diff
@@ -220,26 +215,36 @@ pub fn evaluate_repl(
     Ok(())
 }
 
-fn unescape_for_vscode(text: &mut std::str::Chars) -> Option<char> {
-    match text.next() {
-        Some('\\') => match text.next() {
-            Some('0') => Some('\x00'),  // NUL '\0' (null character)
-            Some('a') => Some('\x07'),  // BEL '\a' (bell)
-            Some('b') => Some('\x08'),  // BS  '\b' (backspace)
-            Some('t') => Some('\x09'),  // HT  '\t' (horizontal tab)
-            Some('n') => Some('\x0a'),  // LF  '\n' (new line)
-            Some('v') => Some('\x0b'),  // VT  '\v' (vertical tab)
-            Some('f') => Some('\x0c'),  // FF  '\f' (form feed)
-            Some('r') => Some('\x0d'),  // CR  '\r' (carriage ret)
-            Some(';') => Some('\x3b'),  // semi-colon
-            Some('\\') => Some('\x5c'), // backslash
-            Some('e') => Some('\x1b'),  // escape
-            Some(c) => Some(c),
-            None => None,
-        },
-        Some(c) => Some(c),
-        None => None,
-    }
+fn escape_special_vscode_bytes(input: &str) -> String {
+    input
+        .chars()
+        .flat_map(|c| {
+            let mut buf = [0; 4]; // Buffer to hold UTF-8 bytes of the character
+            let c_bytes = c.encode_utf8(&mut buf); // Get UTF-8 bytes for the character
+
+            if c_bytes.len() == 1 {
+                let byte = c_bytes.as_bytes()[0];
+
+                match byte {
+                    // Escape bytes below 0x20
+                    b if b < 0x20 => format!("\\x{:02X}", byte).into_bytes(),
+                    // Escape semicolon as \x3B
+                    b';' => "\\x3B".to_string().into_bytes(),
+                    // Escape backslash as \\
+                    b'\\' => "\\\\".to_string().into_bytes(),
+                    // Otherwise, return the character unchanged
+                    _ => vec![byte],
+                }
+            } else {
+                // Escape all multi-byte characters (i.e., those above 0x7F)
+                c_bytes
+                    .bytes()
+                    .flat_map(|b| format!("\\x{:02X}", b).into_bytes())
+                    .collect()
```

Comment:
According to the spec they don't have to be escaped. Not sure if VS Code will reinterpret a multibyte `\xEB\xAB` as utf-8 or expects there to only find lone ASCII `\x7F`

Reply:
ok, i just passed them through now.